src/Sarif/Processors/Log/SarifLogPipeline.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft. All rights reserved. Licensed under the MIT        
+// license. See LICENSE file in the project root for full license information. 
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Newtonsoft.Json;
+
+namespace Microsoft.CodeAnalysis.Sarif.Processors
+{
+    public class ActionTuple
+    {
+        public SarifLogAction Action;
+        public string[] Parameters;
+        
+        /// <summary>
+        /// Two ActionTuples are equal if they take the same actions with the same parameters.
+        /// </summary>
+        public override bool Equals(object obj)
+        {
+            ActionTuple other = obj as ActionTuple;
+            if (other == null)
+            {
+                return base.Equals(obj);
+            }
+            if(this.Action != other.Action)
+            {
+                return false;
+            }
+
+            if (this.Parameters.Length != other.Parameters.Length)
```

Comment:
>this.Parameters [](start = 16, length = 15)

is this.Parameters always guaranteed to be non-null for this instance and the one we're comparing to? If not we will crash